metaflow/meta_files.py

```diff
@@ -0,0 +1,132 @@
+import json
+import os
+
+from enum import Enum
+from typing import Any, Dict, Optional, Union
+
+from .util import get_metaflow_root
+
+_info_file_content = None
+_info_file_present = None
+_included_dist_info = None
+_included_dist_present = None
+
+# Ideally these would be in package/mfenv.py but that would cause imports to fail so
+# moving here. The reason is that this is needed to read extension information which needs
+# to happen before mfenv gets packaged.
+
+MFENV_DIR = (
+    ".mfenv"  # Directory containing "system" code (metaflow and user dependencies)
+)
+MFCONF_DIR = ".mfconf"  # Directory containing Metaflow's configuration files
+MFENV_MARKER = (
+    ".mfenv_install"  # Special file containing metadata about how Metaflow is packaged
+)
+
+
+class MetaFile(Enum):
+    INFO_FILE = "INFO"
+    CONFIG_FILE = "CONFIG_PARAMETERS"
+    INCLUDED_DIST_INFO = "INCLUDED_DIST_INFO"
+
+
+def meta_file_name(name: Union[MetaFile, str]) -> str:
+    if isinstance(name, MetaFile):
+        return name.value
+    return name
+
+
+def generic_get_filename(
+    name: Union[MetaFile, str], is_meta: Optional[bool] = None
+) -> Optional[str]:
+    # We are not in a MFEnv package (so this is an old style package). Everything
+    # is at metaflow root. There is no distinction between code and config.
+    real_name = meta_file_name(name)
+
+    path_to_file = os.path.join(get_metaflow_root(), real_name)
+    if os.path.isfile(path_to_file):
+        return path_to_file
+    return None
+
+
+def v1_get_filename(
+    name: Union[MetaFile, str],
+    meta_info: Dict[str, Any],
+    is_meta: Optional[bool] = None,
```

Comment:
how does this work if i am interested in a file that unfortunately clashes with an internal reserved name? also, is_meta actually useful? what if we simply had another method that provided the contents of `INFO` file and it included everything you need?

Reply:
let me check on that. The idea of putting stuff in mfconf is that they won't clash with code.